### PR TITLE
static authorizer: add e2e tests, additional verifications

### PR DESCRIPTION
**Pull request by s-urbaniak**
This is a followup of #116, adding e2e tests, and verifications around resource requests.

Done: ~TODO: add more e2e tests, especially around parameter rewrites and the static authorizer~

### CODE REVIEW

**Comment by s-urbaniak** [comment]
I think this is ready for a final review cc @jan--f @brancz @standa

**Comment by s-urbaniak** [comment]
Fixes #110

File: examples/static-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
+# Static Authorization example
+
+> Note to try this out with minikube, make sure you enable RBAC correctly. Since minikube v0.26.0 the default bootstrapper is kubeadm - which should enable RBAC by default. For older version follow the instructions [here](../minikube-rbac).
+
+RBAC differentiates in two types, that need to be authorized, resources and non-resources. A resource request authorization, could for example be, that a requesting entity needs to be authorized to perform the `get` action on a particular Kubernetes Deployment.
+
+In this example we deploy the [prometheus-example-app](https://github.com/brancz/prometheus-example-app) and want to preotect it with kube-rbac-proxy, just as detailed in the [rewrite example](../rewrite/README.md). In this example however we will avoid the recurring SubjectAccessReview requests to the api server by allowing kube-rbac-proxy to authorize these requests statically. This is configured in the file passed to the kube-rbac-proxy with the `--config-file` flag. Additionally the `--upstream` flag has to be set to configure the application that should be proxied to on successful authentication as well as authorization.
```
Comment (jan--f) [inline examples/static-auth/README.md:7]:
```suggestion
In this example we deploy the [prometheus-example-app](https://github.com/brancz/prometheus-example-app) and want to protect it with kube-rbac-proxy, just as detailed in the [rewrite example](../rewrite/README.md). In this example however we will avoid the recurring SubjectAccessReview requests to the api server by allowing kube-rbac-proxy to authorize these requests statically. This is configured in the file passed to the kube-rbac-proxy with the `--config-file` flag. Additionally the `--upstream` flag has to be set to configure the application that should be proxied to on successful authentication as well as authorization.
```

**Review by jan--f — APPROVED** [review]
Just a typo, otherwise LGTM

**Comment by s-urbaniak** [comment]
ok, if we don't have any other objections from other maintainers, let's merge :tada:

**Pull request merged**